### PR TITLE
fix: external tooltip legend extra value sync

### LINKED ISSUE #992: External cursor doesn't sync legend values
**Describe the bug**
When connecting an external cursor with the tooltip visibility disabled, the legend extra values do not show.

Related to https://github.com/elastic/elastic-charts/pull/817

**Expected behavior**
The legend values updated based on external cursor events.

**Screenshots**
Screen Recording 2021-01-26 at 11 27 AM

Notice the static legend extra values in the dependent chart.

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/89306

**Checklist**

- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary
fix #992

### Before
Screen Recording 2021-01-26 at 01 04 PM

### After
Screen Recording 2021-01-26 at 12 31 PM

### Checklist

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

File: src/chart_types/xy_chart/state/selectors/get_tooltip_values_highlighted_geoms.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ function getTooltipAndHighlightFromValue(
     return EMPTY_VALUES;
   }
 
-  if (tooltipType === TooltipType.None) {
+  if (tooltipType === TooltipType.None && !externalPointerEvent) {
```
Comment (monfera) [inline src/chart_types/xy_chart/state/selectors/get_tooltip_values_highlighted_geoms.ts:127]: 👍

File: src/components/tooltip/tooltip.tsx [hunk 2]
```diff
@@ -203,9 +203,7 @@ const TooltipComponent = ({
       boundary: boundary === 'chart' && chartRef.current ? chartRef.current : undefined,
     };
   }, [settings, chartRef, rotation]);
-  if (!visible) {
-    return null;
-  }
+
```
Comment (monfera) [inline src/components/tooltip/tooltip.tsx:206]: Nice, adding value by removing code, moreover, a branch! 🌳

**Review by monfera — APPROVED** [review]
LGTM! Wasn't sure why existing mocks gained legends, may not be a bad thing in general

File: src/chart_types/xy_chart/state/selectors/get_tooltip_values_highlighted_geoms.ts (hunk 1 above)
Comment (nickofthyme) [reply src/chart_types/xy_chart/state/selectors/get_tooltip_values_highlighted_geoms.ts:127]: I love when it's just a one-line change, the hard part is knowing what line to change 🧐

**Review by markov00 — APPROVED** [review]
LGTM!

**Pull request merged**